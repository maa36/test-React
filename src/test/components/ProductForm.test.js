import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import ProductForm from '../../components/Products/Update/ProductForm';
import categories from '../../mocks/categories'

test('should test ProductForm component', () => {
  const onSubmitSpy = jest.fn();
   
  const wrapper = shallow(<ProductForm categories={categories} onSave={onSubmitSpy }   />);
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit handler', () => {
  const onSubmitSpy = jest.fn();
  const onSaveSpy = jest.fn();
  const preventDefault = jest.fn();
  const wrapper = shallow(<ProductForm categories={categories} onSave={onSaveSpy}  onSubmit={onSubmitSpy} />);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('Form').simulate('submit',{ preventDefault });
  expect(onSubmitSpy).toHaveBeenCalled();
});

