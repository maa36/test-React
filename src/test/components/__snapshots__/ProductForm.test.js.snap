// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle onSubmit handler 1`] = `
<Form
  onSubmit={[Function]}
  tag="form"
>
  <FormGroup
    tag="div"
  >
    <Label
      for="name"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Name
    </Label>
    <Input
      id="name"
      invalid={true}
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <FormFeedback
      tag="div"
    >
      Name is required, the length must not be greater than 200
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="brand"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Brand
    </Label>
    <Input
      id="brand"
      name="brand"
      onChange={[Function]}
      placeholder="Brand"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="rating"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Rating
    </Label>
    <Input
      id="rating"
      name="rating"
      onChange={[Function]}
      type="select"
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        0
      </option>
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
      <option
        key="6"
        value={6}
      >
        6
      </option>
      <option
        key="7"
        value={7}
      >
        7
      </option>
      <option
        key="8"
        value={8}
      >
        8
      </option>
      <option
        key="9"
        value={9}
      >
        9
      </option>
      <option
        key="10"
        value={10}
      >
        10
      </option>
    </Input>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="categories"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Categories
    </Label>
    <Input
      id="categories"
      invalid={true}
      multiple={true}
      name="categories"
      onChange={[Function]}
      type="select"
      value={Array []}
    >
      <option
        key="1"
        value={1}
      >
        TV & Home Cinema
      </option>
      <option
        key="2"
        value={2}
      >
        Laptops
      </option>
      <option
        key="3"
        value={3}
      >
        Watches
      </option>
      <option
        key="4"
        value={4}
      >
        Headphones
      </option>
      <option
        key="5"
        value={5}
      >
        Phones
      </option>
      <option
        key="6"
        value={6}
      >
        Desktops
      </option>
      <option
        key="7"
        value={7}
      >
        Monitors
      </option>
      <option
        key="8"
        value={8}
      >
        Printers
      </option>
      <option
        key="9"
        value={9}
      >
        Monitors
      </option>
      <option
        key="10"
        value={10}
      >
        Camera & Photo
      </option>
      <option
        key="11"
        value={11}
      >
        Audio & HiFi
      </option>
    </Input>
    <FormFeedback
      tag="div"
    >
      A product must have from 1 to 5 categories
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="itemsInStock"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Items In Stock
    </Label>
    <Input
      id="itemsInStock"
      min="0"
      name="itemsInStock"
      onChange={[Function]}
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="expirationDate"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Expiration date
    </Label>
    <Input
      id="expirationDate"
      invalid={true}
      name="expirationDate"
      onChange={[Function]}
      type="date"
      value=""
    />
    <FormFeedback
      tag="div"
    >
      If a product has an expiration date it must expire not less than 30 days since now
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="receiptDate"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Receipt date
    </Label>
    <Input
      id="receiptDate"
      name="receiptDate"
      onChange={[Function]}
      type="date"
      value=""
    />
  </FormGroup>
  <FormGroup
    check={true}
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
       
      Featured
    </Label>
  </FormGroup>
  <Button
    color="secondary"
    tag="button"
  >
    Submit
  </Button>
</Form>
`;

exports[`should test ProductForm component 1`] = `
<Form
  onSubmit={[Function]}
  tag="form"
>
  <FormGroup
    tag="div"
  >
    <Label
      for="name"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Name
    </Label>
    <Input
      id="name"
      invalid={true}
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <FormFeedback
      tag="div"
    >
      Name is required, the length must not be greater than 200
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="brand"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Brand
    </Label>
    <Input
      id="brand"
      name="brand"
      onChange={[Function]}
      placeholder="Brand"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="rating"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Rating
    </Label>
    <Input
      id="rating"
      name="rating"
      onChange={[Function]}
      type="select"
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        0
      </option>
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
      <option
        key="6"
        value={6}
      >
        6
      </option>
      <option
        key="7"
        value={7}
      >
        7
      </option>
      <option
        key="8"
        value={8}
      >
        8
      </option>
      <option
        key="9"
        value={9}
      >
        9
      </option>
      <option
        key="10"
        value={10}
      >
        10
      </option>
    </Input>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="categories"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Categories
    </Label>
    <Input
      id="categories"
      invalid={true}
      multiple={true}
      name="categories"
      onChange={[Function]}
      type="select"
      value={Array []}
    >
      <option
        key="1"
        value={1}
      >
        TV & Home Cinema
      </option>
      <option
        key="2"
        value={2}
      >
        Laptops
      </option>
      <option
        key="3"
        value={3}
      >
        Watches
      </option>
      <option
        key="4"
        value={4}
      >
        Headphones
      </option>
      <option
        key="5"
        value={5}
      >
        Phones
      </option>
      <option
        key="6"
        value={6}
      >
        Desktops
      </option>
      <option
        key="7"
        value={7}
      >
        Monitors
      </option>
      <option
        key="8"
        value={8}
      >
        Printers
      </option>
      <option
        key="9"
        value={9}
      >
        Monitors
      </option>
      <option
        key="10"
        value={10}
      >
        Camera & Photo
      </option>
      <option
        key="11"
        value={11}
      >
        Audio & HiFi
      </option>
    </Input>
    <FormFeedback
      tag="div"
    >
      A product must have from 1 to 5 categories
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="itemsInStock"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Items In Stock
    </Label>
    <Input
      id="itemsInStock"
      min="0"
      name="itemsInStock"
      onChange={[Function]}
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="expirationDate"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Expiration date
    </Label>
    <Input
      id="expirationDate"
      invalid={true}
      name="expirationDate"
      onChange={[Function]}
      type="date"
      value=""
    />
    <FormFeedback
      tag="div"
    >
      If a product has an expiration date it must expire not less than 30 days since now
    </FormFeedback>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <Label
      for="receiptDate"
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Receipt date
    </Label>
    <Input
      id="receiptDate"
      name="receiptDate"
      onChange={[Function]}
      type="date"
      value=""
    />
  </FormGroup>
  <FormGroup
    check={true}
    tag="div"
  >
    <Label
      check={true}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
       
      Featured
    </Label>
  </FormGroup>
  <Button
    color="secondary"
    tag="button"
  >
    Submit
  </Button>
</Form>
`;
